File: hw/astudillo/U3/Hw19Composite/CompositePython/composite_pattern/model/supervisor.py
from .employee import Employee

class Supervisor(Employee):
    def __init__(self, name=None):
        super().__init__(name)
        self.direct_reports = []
    
    def state_name(self, title, name):
        super().state_name(title, name)
        if self.direct_reports:
            for emp in self.direct_reports:
                emp.state_name(emp.title, emp.name)
    
    def add(self, employee):
        self.direct_reports.append(employee)

File: hw/palacios/u3/HW20Composite/CompositePattern/src/main/java/ec/edu/espe/compositepattern/model/Supervisor.java
package ec.edu.espe.compositepattern.model;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Daniel Palacios
 */
public class Supervisor extends Employee {

    protected List<Employee> directReports = new ArrayList<>();

    public void stateName() {
        super.stateName(); // print name of this employee first
        if (!directReports.isEmpty()) // be sure there are elements
        {
            for (Employee employee : directReports) {
                employee.stateName();
            }
        }
    }

    public void add(Employee anEmployee) {
        this.directReports.add(anEmployee);
    }
}
